drivers/can/can_mcux_flexcan.c

```diff
@@ -91,36 +92,73 @@ struct mcux_flexcan_data {
 	struct mcux_flexcan_tx_callback tx_cbs[MCUX_FLEXCAN_MAX_TX];
 	enum can_state state;
 	can_state_change_isr_t state_change_isr;
+	flexcan_timing_config_t timing;
 };
 
-static int mcux_flexcan_configure(const struct device *dev,
-				  enum can_mode mode,
-				  uint32_t bitrate)
+static uint32_t mcux_flexcan_get_core_clock(const struct device *dev)
 {
 	const struct mcux_flexcan_config *config = dev->config;
-	flexcan_config_t flexcan_config;
 	const struct device *clock_dev;
 	uint32_t clock_freq;
 
 	clock_dev = device_get_binding(config->clock_name);
 	if (clock_dev == NULL) {
-		return -EINVAL;
+		return 0;
 	}
 
 	if (clock_control_get_rate(clock_dev, config->clock_subsys,
 				   &clock_freq)) {
+		return 0;
+	}
+
+	return clock_freq;
+}
+
+static int mcux_flexcan_set_timing(const struct device *dev,
+				   const struct can_timing *timing,
+				   const struct can_timing *timing_data)
+{
+	struct mcux_flexcan_data *data = dev->data;
+	const struct mcux_flexcan_config *config = dev->config;
+
+	if (!timing) {
 		return -EINVAL;
 	}
 
+	data->timing.preDivider = timing->prescaler;
+	data->timing.rJumpwidth = timing->sjw;
+	data->timing.phaseSeg1 = timing->phase_seg1;
+	data->timing.phaseSeg2 = timing->phase_seg2;
+	data->timing.propSeg = timing->prop_seg;
+
+	FLEXCAN_SetTimingConfig(config->base, &data->timing);
+
+	return 0;
+}
+
+static int mcux_flexcan_set_mode(const struct device *dev, enum can_mode mode)
+{
+	struct mcux_flexcan_data *data = dev->data;
+	const struct mcux_flexcan_config *config = dev->config;
+	flexcan_config_t flexcan_config;
```

Comment:
```suggestion
```
I think you can update `data->timing` instead and also pass it to `FLEXCAN_Init` at the end. Requires less updates of member variables below, as they are already in the correct state. And it preserves the mode if timing is changed afterwards.

Reply:
Not sure what you meant here

Reply:
Oops, never mind. I confused `flexcan_config_t` and `flexcan_timing_config_t` and thought that the timing parameter of the `mcux_flexcan_data` could be re-used here. But looking twice it doesn't make sense.